core/src/service/sync_service.rs

```diff
@@ -355,11 +346,11 @@ impl<Client: Requester> RequestContext<'_, '_, Client> {
         }
 
         // base = local
-        self.tx
-            .base_metadata
+        (&mut self.tx.base_metadata)
             .to_lazy()
             .stage(local_changes_no_digests)
             .promote();
+        self.cleanup_local_metadata();
```

Comment:
(here and elsewhere) staging changes on local and promoting will automatically keep local and base duplicate-free, but staging changes on base and promoting does not (it would somehow need to reference local to know how). It used to be that the act of staging local on base would prune duplicates (which oddly could cause e.g. a call to `calculate_work` to change files on disk) but now we explicitly prune after promoting any changes to base which happens in `pull`, `push_metadata`, and `push_documents`